log/logClient.go

```diff
@@ -192,8 +193,21 @@ func (fd *Feeder) WatchMessages(msgPath string, jsonFormat bool) error {
 	return nil
 }
 
+func RegexChecker(filterstring string, res string) (bool, error) {
```

Comment:
```suggestion
func RegexMatcher(filter string, res string, caseSensitive bool) (bool, error) {
```
I think it would be better to add the case insensitivity string to filter as part of the function only

Reply:
yeah, we can do that but I think adding the `caseSensitive` parameter to the function will add some code duplication inside the `RegexMatcher`  function  ...
the function will look like
```
func RegexMatcher(filter string, res string, caseSensitive bool)(bool , error){

        if !caseSensitive{
	
	re, err := regexp.Compile(filterstring)
	if err != nil {
		return false, err
	}
	match := re.MatchString(res)
	if !match {
		return false, nil
	}
	return true, nil

	} else{

	re, err := regexp.Compile("(?i)"+ filterstring)
	if err != nil {
		return false, err
	}
	match := re.MatchString(res)
	if !match {
		return false, nil
	}
	return true, nil
	}

}
```
rather than for the current Implementation, we only need to add the two extra strings for the `logType` and `operation` field.WDYT?

Reply:
```
if caseSensitive {
 filter = "(?i)"+filter
}
... // Rest of the function remains the same
```

I believe this should be enough, no need to have so much duplication🤔